tutorials/run-laravel-on-appengine-standard/index.md

```diff
@@ -89,19 +102,149 @@ from laravel.com.
+        |
+        | Before we start up Laravel, let's check if we are running in GCP App Engine,
+        | and, if so, make sure that the Laravel's storage directory structure is
+        | present.
+        |
+        */
+
+        if (getenv('GCP_APP_ENGINE_LARAVEL')
+            && !file_exists('/tmp/.dirs_created')) {
+            foreach (['/tmp/app/public',
+                        '/tmp/framework/cache/data',
+                        '/tmp/framework/sessions',
+                        '/tmp/framework/testing',
+                        '/tmp/framework/views',
+                        '/tmp/logs'] as $tmpdir) {
+                if (!file_exists($tmpdir)) {
+                    mkdir($tmpdir, 0755, true);
+                }
+            }
+            touch('/tmp/.dirs_created');
+        }
+        # [END]
     
-        composer remove --dev facade/ignition
+    This code is where the `GCP_APP_ENGINE_LARAVEL` environment variable from 
+    `app.yaml` comes into play. This avoids running this code in local 
+    development environment.
+
+    If your code writes into any storage directory, you should make sure to 
+    create the necessary directory structure here.
+
+1.  Modify `composer.json` file by adding a few scripts scripts. Here is an 
+    example result, with the additions detailed below:
+
+        "scripts": {
+            "post-autoload-dump": [
+                "Illuminate\\Foundation\\ComposerScripts::postAutoloadDump",
+                "mkdir -p bootstrap/cache",
+                "@php artisan package:discover --ansi"
+            ],
+            "post-root-package-install": [
+                "@php -r \"file_exists('.env') || copy('.env.example', '.env');\""
+            ],
+            "post-create-project-cmd": [
+                "@php artisan key:generate --ansi"
+            ],
+            "gcp-build": "sed -i -e \"s|env('APP_STORAGE', base_path() . '/storage')|'/tmp'|g\" bootstrap/app.php && mkdir -p /tmp/framework/views && mv .env.gae .env && php artisan config:cache && rm -f .env && php artisan route:cache"
+        }
+    
+    In `"scripts"` -> `"post-autoload-dump"` add the following script right 
+    before `package:discover` command to ensure that the command has the 
+    directory to write into:
+
+        ...
+        "mkdir -p bootstrap/cache",
+        ...
+    
+    In `"scripts"` add the `"gcp-build"` field, which is a (currently 
+    undocumented) way of running custom logic during deployment to App Engine.
+
+        ...
+        "gcp-build": "sed -i -e \"s|env('APP_STORAGE', base_path() . '/storage')|'/tmp'|g\" bootstrap/app.php && mkdir -p /tmp/framework/views && mv .env.gae .env && php artisan config:cache && rm -f .env && php artisan route:cache"
+        ...
+    
+    Unfortunately at this time App Engine does not support an array of commands 
+    under `"gcp-build"`. Here are the details on the chained commands:
```

Comment:
That's interesting, as this should be executed by `composer`, which _does_ support it. How is it that it is not supported?

Reply:
Came as a surprise for me too. Whenever I put an array there, the build goes down with `cannot unmarshal array into Go struct field`.